# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/Layers.cpp
# Commit: 77ef8e8bc669
# Full Hash: 77ef8e8bc669c9ecab4d94505b11b3cb732446c5
# Author: Alexis Beingessner <a.beingessner@gmail.com>
# Date: 2019-07-15 21:39:46
# Regressor Bug: 1547351
# File Overlap Count: 2
# Description:
#   Bug 1547351 - Automatically set content process render root r=kats
#   
#   This implements the idea of automatically setting a content proc's
#   render root based on the render root enclosing the iframe that
#   points to it. There was a bit of cleanup in here that was a bit
# ==============================================================================

diff -r b594f8e47025 -r 77ef8e8bc669 gfx/layers/Layers.cpp
--- a/gfx/layers/Layers.cpp	Mon Jul 15 13:20:31 2019 +0000
+++ b/gfx/layers/Layers.cpp	Mon Jul 15 13:18:44 2019 +0000
@@ -2240,7 +2240,8 @@
   wr::RenderRoot renderRoot = (GetBackendType() == LayersBackend::LAYERS_WR)
                                   ? aRenderRoot
                                   : wr::RenderRoot::Default;
-  mPendingScrollUpdates[renderRoot][aScrollId] = aUpdateInfo;
+  bool ok = mPendingScrollUpdates[renderRoot].put(aScrollId, aUpdateInfo);
+  MOZ_RELEASE_ASSERT(ok);
   return true;
 }
 
@@ -2249,11 +2250,8 @@
   // This never gets called for WebRenderLayerManager, so we assume that all
   // pending scroll info updates are stored under the default RenderRoot.
   MOZ_ASSERT(GetBackendType() != LayersBackend::LAYERS_WR);
-  auto it = mPendingScrollUpdates[wr::RenderRoot::Default].find(aScrollId);
-  if (it != mPendingScrollUpdates[wr::RenderRoot::Default].end()) {
-    return Some(it->second);
-  }
-  return Nothing();
+  auto p = mPendingScrollUpdates[wr::RenderRoot::Default].lookup(aScrollId);
+  return p ? Some(p->value()) : Nothing();
 }
 
 std::unordered_set<ScrollableLayerGuid::ViewID>
@@ -2261,8 +2259,8 @@
   std::unordered_set<ScrollableLayerGuid::ViewID> scrollIds;
   for (auto renderRoot : wr::kRenderRoots) {
     auto& updates = mPendingScrollUpdates[renderRoot];
-    for (const auto& update : updates) {
-      scrollIds.insert(update.first);
+    for (auto it = updates.iter(); !it.done(); it.next()) {
+      scrollIds.insert(it.get().key());
     }
     updates.clear();
   }